feat(Settings/Communities): implement new communities list

caybro opened the pull request:
Implements the new Settings/Communities view according to the Figma design: https://www.figma.com/file/idUoxN7OIW2Jpp3PMJ1Rl8/%E2%9A%99%EF%B8%8F-Settings-%7C-Desktop?type=design&node-id=16089-387522&mode=design&t=mt6XicISfqBzHVqx-0

- adds the corresponding storybook page to play around with various types of communities
- exchanges the mute/unmute menu icons according to the new design
- TokenMaster category TODO

Closes #11145

### Affected areas

Settings/Communities

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

Admin:
![image](https://github.com/status-im/status-desktop/assets/5377645/c12e4ca6-24f5-4c81-8c24-fb05c1ce0223)

Member:
![image](https://github.com/status-im/status-desktop/assets/5377645/36fcbc2a-7e29-4fc9-a0d3-55249989d38a)

Pending:
![image](https://github.com/status-im/status-desktop/assets/5377645/5da8cf8c-12f1-4bcf-b324-b44f992a96c9)

Storybook:
![image](https://github.com/status-im/status-desktop/assets/5377645/e84836e7-ccd2-4122-8576-82626c1db6d9)

Storybook, empty communities:
![image](https://github.com/status-im/status-desktop/assets/5377645/5a714ec2-d3fe-4f93-aca7-dee206a4cd1f)

caybro commented on ui/app/AppLayouts/Profile/views/CommunitiesView.qml:
```diff
@@ -90,52 +89,136 @@ SettingsContentBase {
+
+    readonly property var communityIntroDialogPopup: Component {
+        id: communityIntroDialogPopup
+        CommunityIntroDialog {
```
This `CommunityIntroDialog` should probably get moved to the shared `Popups.qml`

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | a6a6dc1f | [#1](https://ci.status.im/job/status-desktop/job/branches/job/linux/job/x86_64/job/tests-imports/job/PR-11214/1/) | 2023-06-21 20:44:50 | ~5 min | `tests/imports` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/branches/job/linux/job/x86_64/job/tests-imports/job/PR-11214/1//consoleText) |

caybro commented on ui/app/AppLayouts/Chat/ChatLayout.qml:
```diff
@@ -91,25 +91,6 @@ StackLayout {
                 }
             }
-
-            CommunityIntroDialog {
```
This one is a duplicate; already present and used as a `Component`

0x-r4bbit replied:
I've introduced this one here in `ChatLayout` so that it's available to be spun up by `ChatView` or `JoinCommunityView`.
We gotta make sure that this doesn't break with this removal.

Just raising this here...

caybro replied:
I just double checked, so:

- leaving a random community from the primary nav: 
![image](https://github.com/status-im/status-desktop/assets/5377645/8f811f40-8141-413e-8a46-d2c7a3a9b984)

- doing the same from Settings/Communities:
![image](https://github.com/status-im/status-desktop/assets/5377645/0a54538f-a6bb-4a53-bcf0-e534c20dc0a6)

0x-r4bbit replied:
These are not the intro dialogs though. 
The intro dialogs are shown when you want to request access to a community, right before you enter your password to reveal your addresses.

caybro replied:
Yeah it's the same dialog, really :) Just has a different set of footer buttons based on the params you set. But the point was that the instance I removed was not being used at all in the existing `ChatLayout.qml`. 

There is:
```qml
Component {
        id: communityIntroDialogPopup
        CommunityIntroDialog {
            id: communityIntroDialog

            property string communityId
...
}
```
which is used as:
```qml
            onRevealAddressClicked: {
                Global.openPopup(communityIntroDialogPopup, {
                    communityId: communityData.id,
                    isInvitationPending: joinCommunityView.isInvitationPending,
                    name: communityData.name,
                    introMessage: communityData.introMessage,
                    imageSrc: communityData.image,
                    accessType: communityData.access
                })
            }
```

caybro replied:
Ah yes, it's this dialog, still working:

![image](https://github.com/status-im/status-desktop/assets/5377645/224b8f6b-0111-4ac6-ab1c-b2790c9f0a91)

and

![image](https://github.com/status-im/status-desktop/assets/5377645/9ac9e5cf-5be4-40aa-a8f1-5fd1d9eb43fc)

0x-r4bbit commented on ui/app/AppLayouts/Chat/panels/communities/CommunityMembersSettingsPanel.qml:
```diff
@@ -108,11 +108,10 @@ SettingsPageLayout {
                 rootStore: root.rootStore
                 placeholderText: {
-                    if (root.membersModel.count === 0) {
+                    if (root.membersModel.count === 0)
```
Can leave as is here but just wanna point out that it's rather bad practice to remove these parens in JS, even if the resulting statement is a single line

caybro replied:
Yeah I see... it's a common practice in C++ to have single line statements w/o the braces

0x-r4bbit reviewed (approved):
Based on screenshots this LGTM.
I haven't tested this locally though.

Left some comments in the code. I think the popup one is rather important.

Great job

caybro commented:
Since for some reason I can't reply to this comment inline:

> I've introduced this one here in ChatLayout so that it's available to be spun up by ChatView or JoinCommunityView.
We gotta make sure that this doesn't break with this removal.
> Just raising this here...

@0x-r4bbit the CommunityIntroDialog is there already, see https://github.com/status-im/status-desktop/blob/e24e1ab768ea2eba71e637273602bd8ee5d2e508/ui/app/AppLayouts/Chat/ChatLayout.qml#L172-L192 and is throughout the file for this purpose. The one I removed was not used at all

borismelnik reviewed (approved):
👍 looks perfect

osmaczko commented on ui/app/AppLayouts/Profile/panels/CommunitiesListPanel.qml:
```diff
@@ -8,88 +8,155 @@ import StatusQ.Controls 0.1
     signal setCommunityMutedClicked(string communityId, int mutedType)
-
     signal setActiveCommunityClicked(string communityId)
+    signal showCommunityIntroDialog(string communityId, string name, string introMessage, string imageSrc, int accessType)
+    signal cancelMembershipRequest(string communityId)
```
By name, it appears as a function from the client's side. It would be great if this followed the Qt signal naming convention.

caybro replied:
Agree... but existing code :)

osmaczko commented on ui/app/AppLayouts/Profile/views/CommunitiesView.qml:
```diff
@@ -90,52 +89,140 @@ SettingsContentBase {
+        color: Theme.palette.baseColor1
+    }
+
+    component Panel: CommunitiesListPanel {
```
:+1:
